finetune/scheduler.py

```diff
@@ -188,10 +190,27 @@ def in_cache(self, cache_key, key):
             return cache_key in self.loaded_models
         return f"{cache_key}_key={key}" in self.loaded_models
 
-    @functools.lru_cache
-    def etl_cache(self, model_file_path):
-        return SequenceLabeler.load(model_file_path, key="etl")
+    def load_etl(self, model_file_path, cache_key):
+        if cache_key in self.etl_cache:
+            etl = self.etl_cache.get(cache_key)
+        else:
+            etl = SequenceLabeler.load(model_file_path, key="etl")
+            self.etl_cache[cache_key] = etl
+        return etl
+
+
+class EtlCache:
+    def __init__(self, maxsize=128):
+        self.cache = OrderedDict()
+        self.maxsize = maxsize
+
+    def __set__(self, key, value):
+        if len(self.cache) >= self.maxsize:
+            self.cache.popitem(last=False)
+        self.cache[key] = value
+
+    def get(self, key):
+        return self.cache.get(key, None)
 
-    def etl_in_cache(self, cache_key, key="etl"):
```

Comment:
We don't still need the etl_in_cache method?

Reply:
we had built our own etl_cache instead and moved it directly on to the scheduler